table/tables/tables.go

```diff
@@ -1024,3 +1027,118 @@ func init() {
 	table.TableFromMeta = TableFromMeta
 	table.MockTableFromMeta = MockTableFromMeta
 }
+
+// ctxForPartitionExpr implement sessionctx.Context interfact.
+type ctxForPartitionExpr struct {
+	sessionVars *variable.SessionVars
+}
+
+// newCtxForPartitionExpr creates a new sessionctx.Context.
+func newCtxForPartitionExpr() sessionctx.Context {
+	sctx := &ctxForPartitionExpr{
+		sessionVars: variable.NewSessionVars(),
+	}
+	sctx.sessionVars.MaxChunkSize = 2
+	sctx.sessionVars.StmtCtx.TimeZone = time.UTC
```

Comment:
Why use UTC here? Could we set it to the value of time_zone system variable?

Reply:
Both are not ideal, it should be user's session timezone, but we can't obtain it.

Reply:
If Expression use this ctx when creating the table, rather than the ctx when evaluating partition expression, we'll always meet some bugs, no matter it's UTC or Local here. @shenli

Reply:
Maybe we should remove the `ctx` field from expression firstly.

Reply:
So, merge this PR until @XuHuaiyu finish that?

Reply:
If `Expression` use the mock `ctx` when creating the table, we still can meet some bugs, So the best way to make the query on a partitioned table work is to remove the `ctx` out of expression. Or we store the expression string and parse it to expression every time we parse a query on a partitioned table.

Reply:
So, what is the conclusion about this? Should we file an issue to trace it?